lib/puppet/resource.rb

```diff
@@ -633,6 +561,12 @@ def validate_parameter(name)
     raise Puppet::ParseError.new(_("no parameter named '%{name}'") % { name: name }, file, line) unless valid_parameter?(name)
   end
 
+  # This method, together with #file and #line, makes it possible for a Resource to be a 'source_pos' in a reported issue.
+  # @return [Integer] Instances of this class will always return 0.
+  def pos
+    0
+  end
```

Comment:
IIRC, the reporting looks at file, line and pos, and determines based on what is set how it will format the location (with or without line and pos depending on if they are `nil` or not).